aws/rust-runtime/aws-http/src/auth.rs

```diff
@@ -186,14 +183,19 @@ mod tests {
     #[tokio::test]
     async fn async_map_request_apply_populates_credentials() {
         let mut req = operation::Request::new(http::Request::new(SdkBody::from("some body")));
-        set_provider(
+        let credentials_cache = CredentialsCache::lazy_builder()
+            .into_credentials_cache()
+            .create_cache(Arc::new(provide_credentials_fn(|| async {
+                Ok(Credentials::new("test", "test", None, None, "test"))
```

Comment:
I'm thinking we should consider adding a `test` method to `Credentials` with a `#[cfg(test)]` gate. The function would return these credentials or similar.

Reply:
That's good idea. I'll tackle it in a separate PR as it will probably affect many call sites currently creating a test `Credentials`.

Reply:
[PR](https://github.com/awslabs/smithy-rs/pull/2145) created.